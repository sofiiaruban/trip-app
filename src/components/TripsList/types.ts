export interface TripProps {
  cityImgSrc: string
  cityName: string
  startDate: Date
  endDate: Date
}
