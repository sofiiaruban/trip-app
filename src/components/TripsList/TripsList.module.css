.tripsListContainer {
  display: flex;
  gap: 1rem;
}
.tripsList {
  list-style-type: none;
  margin: 0;
  padding: 0;
  display: flex;
  gap: 1rem;
  overflow-x: scroll;
  overflow-y: hidden;
  width: 50%;
  padding: 1rem;
}
.addTripItem {
  background-color: rgba(128, 128, 128, 0.6);
  width: 120px;
  height: 50%;
  display: flex;
  justify-content: center;
  align-items: center;
  font-weight: 600;
  padding: 2rem 2rem;
  margin: 1rem;
}
.addTripTitle {
  font-weight: 600;
  font-size: medium;
}
