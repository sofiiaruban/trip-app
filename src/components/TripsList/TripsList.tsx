import { FC } from 'react'
import TripItem from './TripItem/TripItem'
import { Trip } from '../../types'
import styles from './TripsList.module.css'
import ButtonIcon from '../Buttons/ButtonIcon'
import { FaPlus } from 'react-icons/fa6'

interface TripsProps {
  trips: Trip[]
}
const TripsList: FC<TripsProps> = ({ trips }) => {
  return (
    <ul className={styles.tripsList}>
      {trips.map((trip) => (
        <li key={trip.cityName}>
          <TripItem
            cityImgSrc={trip.cityImgSrc}
            cityName={trip.cityName}
            startDate={trip.startDate}
            endDate={trip.endDate}
          />
        </li>
      ))}
      <li className={styles.addTripItem}>
        <ButtonIcon>
          <FaPlus />
          <p className={styles.addTripTitle}>Add trip</p>
        </ButtonIcon>
      </li>
    </ul>
  )
}

export default TripsList
