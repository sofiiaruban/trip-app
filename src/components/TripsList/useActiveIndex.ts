import { useState } from 'react';

const useActiveIndex = (initialIndex: number = 0) => {
  const [activeIndex, setActiveIndex] = useState(initialIndex);

  const setActive = (index: number) => {
    setActiveIndex(index);
  };

  return { activeIndex, setActive };
};

export default useActiveIndex;
