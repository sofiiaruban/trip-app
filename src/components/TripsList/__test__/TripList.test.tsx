import { render } from '@testing-library/react'
import TripsList from '../TripsList'

test('TripsList matches snapshot', () => {
  const setIsActiveMock = jest.fn()
  const commonProps = {
    trips: [
      {
        setIsActive: setIsActiveMock,
        cityImgSrc: '/src',
        cityName: 'KYIV',
        startDate: new Date('24-10-01'),
        endDate: new Date('24-10-03')
      }
    ],
    activeIndex: 0,
    setActive: setIsActiveMock
  }

  const { container } = render(<TripsList {...commonProps} />)

  expect(container).toMatchSnapshot()
})
