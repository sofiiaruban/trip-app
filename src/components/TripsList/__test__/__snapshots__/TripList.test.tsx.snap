// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TripsList matches snapshot 1`] = `
<div>
  <div
    class="tripsListContainer"
  >
    <ul
      class="tripsList"
    >
      <li>
        <article
          class="trip active"
        >
          <img
            alt="KYIV city"
            class="cityImg"
            src="/src"
          />
          <div
            class="tripInfo"
          >
            <p
              class="cityName"
            >
              KYIV
            </p>
            <p
              class="tripDates"
            >
              <span>
                Invalid Date
              </span>
               - 
              <span>
                Invalid Date
              </span>
            </p>
          </div>
        </article>
      </li>
    </ul>
    <div
      class="addTripItem"
    >
      <button
        class="buttonIcon"
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
          />
        </svg>
        <p
          class="addTripTitle"
        >
          Add trip
        </p>
      </button>
    </div>
  </div>
</div>
`;
